### ISSUE TITLE
Adding ContainerMargins gives out constraint errors

### ISSUE THREAD

**Issue by ag-hendrix** [SCENEE/FloatingPanel#355]
### Description

Hello! I have a custom class called FloatingPanelBottomSheet which extends FloatingPanelController. I'm using it to present a bottom sheet (SingleChoiceVC with custom view SingleChoiceView) that has a title and a tableview which the user can select from.  SingleChoiceView is all coded programmatically and uses Auto Layout.

I was able to set the containerMargins to .init(top: 20, left: 0, bottom: 0, right: 0) so that the grabber floats outside of the floating panel.

### Expected behavior

I'm getting the expect behaviour without constraint errors.

### Actual behavior

I'm getting the expect behaviour with constraint errors.

ios[31738:842055] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x6000019ddcc0 ios.SingleChoiceView:0x7f9e650c0db0.height == FloatingPanel.FloatingPanelSurfaceView:0x7f9e62409910.height - 20   (active)>",
    "<NSLayoutConstraint:0x6000019e0aa0 FloatingPanel.FloatingPanelSurfaceView:0x7f9e62409910.bottom == FloatingPanel.FloatingPanelPassThroughView:0x7f9e650c35f0.bottom   (active)>",
    "<NSLayoutConstraint:0x6000019e3160 V:[FloatingPanel.FloatingPanelPassThroughView:0x7f9e650c35f0]-(0)-[FloatingPanel.FloatingPanelSurfaceView:0x7f9e62409910]   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x6000019ddcc0 ios.SingleChoiceView:0x7f9e650c0db0.height == FloatingPanel.FloatingPanelSurfaceView:0x7f9e62409910.height - 20   (active)>

It tried setting the contentViewController to a new instance of UIViewController() to see if I would get the same constraint errors and it does.

Is there any way to fix this?

Thank you.

**How do you display panel(s)?**

I'm presenting the panel modally.

**How many panels do you displays?**

I'm displaying 1 panel.

### Environment

**Library version**

FloatingPanel (1.7.4)

**Installation method**

* CocoaPods
 
**iOS version(s)**

iOS 13.2

**Xcode version**

11.2

**Comment by SCENEE** [comment]
You seem to use a layout object adopting `FloatingPanelIntrinsicLayout` because the constraint error is caused by the below constraint.
```
"<NSLayoutConstraint:0x6000019e3160 V:[FloatingPanel.FloatingPanelPassThroughView:0x7f9e650c35f0]-(0)-[FloatingPanel.FloatingPanelSurfaceView:0x7f9e62409910] (active)>"
```
(It means "FloatingPanelSurfaceView's top edge should equal FloatingPanelPassThroughView's bottom edge.")

I guess the constant errors will be fixed using a layout adopting `FloatingPanelLayout`. For your quick check, you can returns `FloatingPanelDefaultLayout()` object in `floatingPanel(_:layoutFor:) delegate method` :relieved:

**Comment by ag-hendrix** [comment]
Thank you for the quick response! 

I tried returning the `FloatingPanelDefaultLayout()` object in `floatingPanel(_:layoutFor:)` delegate method and I'm still getting the same constraint error.  It's as if the constraints aren't properly updated when we set `.containerMargins`.  Could there be another way to have the grabber outside the floating panel without setting `.containerMargins`?

**Comment by SCENEE** [comment]
@ag-hendrix 
This might be fixed on v1.7.5 by this patch #359. And also you have another option to have the grabber outside of a pane to set a minus value to `.grabberTopPadding`.

```swift
fpc.surfaceView.grabberTopPadding = -20
```

**Comment by ag-hendrix** [comment]
@SCENEE 
It's fixed now on v1.7.5! Thank you so much!! 😄
